yt_dlp/utils.py

```diff
@@ -6307,6 +6307,12 @@ def calculate_preference(self, format):
         # if format.get('preference') is None and format.get('ext') in ('f4f', 'f4m'):  # Not supported?
         #    format['preference'] = -1000
 
+        if format.get('preference') is None and format.get('ext') == 'flv' and format.get('vcodec') in ('hevc', 'h265'):
+            # HEVC-over-FLV is out-of-spec by FLV's original spec
+            # ref. https://trac.ffmpeg.org/ticket/6389
+            # ref. https://github.com/yt-dlp/yt-dlp/pull/5821
+            format['preference'] = -1000
```

Comment:
Note: -1000 means the format won't be visible in `-F`

Reply:
```
$ ./yt-dlp.sh -F https://live.bilibili.com/6 
[BiliLive] Extracting URL: https://live.bilibili.com/6
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[BiliLive] 6: Downloading JSON metadata
[info] Available formats for 6:
ID               EXT RESOLUTION │ PROTO │ VCODEC ACODEC  MORE INFO
──────────────────────────────────────────────────────────────────
ultra_high_res-0 flv unknown    │ https │ hevc   unknown 超清
ultra_high_res-1 flv unknown    │ https │ hevc   unknown 超清
source-0         flv unknown    │ https │ hevc   unknown 原画
source-1         flv unknown    │ https │ hevc   unknown 原画
high_res-0       flv unknown    │ https │ avc    unknown 高清
high_res-1       flv unknown    │ https │ avc    unknown 高清
ultra_high_res-2 flv unknown    │ https │ avc    unknown 超清
ultra_high_res-3 flv unknown    │ https │ avc    unknown 超清
source-2         flv unknown    │ https │ avc    unknown 原画
source-3         flv unknown    │ https │ avc    unknown 原画
```
I do see it now - an another bug?

Reply:
hm.. -1000 itself is shown, apparently. https://github.com/yt-dlp/yt-dlp/blob/c73355510629e3eda5a79d4e2876a35316ca6ed2/yt_dlp/YoutubeDL.py#L3623

I don't know whether this is intentional or a bug. I actually don't know of any extractor that uses this feature or even why this exists...

Reply:
Anyway I changed it to -100 since there's no point in forbidding from downloading